scripts/languageserver/provider_misc.jl

```diff
@@ -75,10 +75,36 @@ end
 abstract didChange <: Method
 
 function Respond(r::Request{didChange,DidChangeTextDocumentParams})
+        vtdi = r.params.textDocument
+        C = r.params.contentChanges
+        for c in C
+            c(vtdi)
+        end
     try
-        documents[r.params.textDocument.uri] = split(r.params.contentChanges[1].text, r"\r\n?|\n")
         return
     catch err
         return Response{didChange,Exception}("2.0",r.id,err)
     end
 end
+
+function (c::TextDocumentContentChangeEvent)(vtdi::VersionedTextDocumentIdentifier)
+    d = join(documents[vtdi.uri],'\n').data
```

Comment:
I don't know much about String handling tbh, I've always had the impression it's all quite slow in Julia?. I'm going to do this properly (tediously) and set up and compare 4 approaches (full sync, Vector{UInt},, Vector{String} and Vector{SubString})